src/App.jsx

```diff
@@ -45,14 +59,25 @@ class App extends Component {
   }
 
   runPostLogin() {
-    this.setState({ isLoggedIn: true });
-    GitHubService.getFile(this.props.config.repo).then((source) => {
-      DocumentService.setSource(source);
-      this.setState({ content: DocumentService.getJSON() });
+    this.setState({
+      isLoggedIn: true,
+      isLoading: true,
+    });
+    GitHubService.loadSchema(this.props.config).then((schema) => {
+      store.dispatch({ type: 'SET_SCHEMA', value: schema });
+      GitHubService.getFile(this.props.config.repo).then((source) => {
+        DocumentService.setSource(source);
```

Comment:
I'd instead update the Document service from the reducer so you don't have set it twice, would that work in this case?